Writerside/topics/results-and-discussion/testing.md

```diff
@@ -0,0 +1,42 @@
+# Testing
+
+In general, tests are divided into unit test cases and integration test cases. Unit tests specifically target one unit
+of code (typically a class) in isolation, often with mocked dependencies when necessary. Integration tests, on the other
+hand, assess multiple components interacting together (e.g., a file reader working with a records parser and test data).
+In our case, integration tests may also include running external dependencies, such as in the client, where a service
+instance must be started using Docker Compose. We acknowledge that the boundary between integration and system tests can
+be blurred, but we have chosen to categorize these as integration tests rather than introducing a third test category.
+It is important that unit tests and integration tests can be executed in distinct phases.
+
+We did not prioritize achieving the highest possible line coverage. Instead, we used this metric as a guiding tool
+during development. The focus was on testing the most relevant and complex features. During the review process,
+reviewers were required to check the code against the [Code Style Guide](code-style-guide.md), which also included
+ensuring that new or modified components were properly tested.
```

Comment:
I would maybe try highlighting this more. Because I think it's important to understand that we tried to not develop features with no tests and write tests sometimes later.

Reply:
https://github.com/naviqore/documentation/pull/25/commits/f395a3d852abf63e0c1a458e432477ae70524c28